rust/src/ketama.rs

```diff
@@ -2,18 +2,20 @@ use md5::{Digest, Md5};
 
 pub struct KetamaPool {
     /// The list of Servers, sorted by their ranking value.
-    ranking: Vec<ServerRank>,
+    pub ranking: Vec<ServerRank>,
 }
 
-struct ServerRank {
-    value: u32,
-    index: u32,
+pub struct ServerRank {
+    pub server_name: String,
+    pub value: u32,
+    pub index: u32,
 }
 
 impl KetamaPool {
     /// Builds a new pool using the given hash keys.
     pub fn new(keys: &[&str]) -> Self {
-        let ranking = create_server_ranking(keys);
+        let mut ranking = Vec::with_capacity(POINTS_PER_SERVER * POINTS_PER_HASH * keys.len());
```

Comment:
the C implementation reserves a bit of slack space at the end, not quite sure if we want to do this, as an `add_node` call would essentially double the capacity with the default growth factor.